React/Tailwind/Best practices.md

```diff
@@ -0,0 +1,272 @@
+
+```css
+@utility {
+  .clip-blob {
+    clip-path: polygon(50% 0%, 100% 25%, 100% 75%, 50% 100%, 0% 75%, 0% 25%);
+  }
+}
+```
+
+You can now reuse `clip-blob` alongside built-in utilities with complete purge safety and no JavaScript.
+
+### How does the group utility help with state-based styling?
+
+Instead of repeating the same prefix on multiple siblings (`hover:`, `focus:`), wrap them in a parent that carries `group`:
+
+```tsx
+<button className="group inline-flex items-center gap-2 p-3 rounded-lg bg-brand-600">
+  <Icon className="size-4 text-white group-hover:translate-x-0.5 transition-transform" />
+  <span className="text-white group-hover:underline">Download</span>
+</button>
+```
+
+Any child can now react to `group-hover`, `group-focus`, `group-aria-expanded`, etc., keeping class strings short and readable.
+
+### My hover and focus class lists are gigantic. Is there a cleaner pattern?
+
+Use CVA or the `cn` helper to build a map of state variants instead of stacking prefixes manually:
+
+```tsx
+const card = cva("rounded-lg p-4 transition", {
+  variants: {
+    intent: {
+      default: "",
+      hover: "hover:bg-brand-50 hover:shadow",
+      focus: "focus-visible:ring-2 focus-visible:ring-brand-600",
+    },
+  },
+});
+```
+
+Apply the variant you need with `card({ intent: 'hover' })`. Literal strings remain visible for the compiler and you avoid prefix noise.
+
+### How do I configure Material UI so Tailwind utilities take effect?
+
+In MUI v5 you must enable CSS custom properties:
+
+```tsx
+import { createTheme } from "@mui/material/styles";
+
+const theme = createTheme({
+  cssVarPrefix: "mui", // any prefix is fine
+  cssVarEnabled: true, // critical: ensures Tailwind variables cascade
+});
+```
+
+With `cssVarEnabled` set to true, Tailwind classes applied via `className` or `sx` override MUIs default palette and spacing. Remember to load `tailwind.css` after the MUI baseline styles so utility specificity wins.
+
+### When is it acceptable to sprinkle !important?
+
+Only when a third-party library injects an inline style or a high-specificity rule that you cannot control. Scope the override to the smallest selector possible, document the reason, and keep a TODO to remove it once the upstream lib allows custom classes.
+
+### I keep typing align-center instead of items-center. How do I memorise Tailwind's naming?
+
+Tailwind mirrors **justify-** for main-axis alignment and **items-** for cross-axis. A quick mnemonic: “Flex items cross the axis”. Also make sure to turn on IntelliSense - misspelled utilities show a red underline and the correct suggestion in the hover tooltip.
+
+If you really can't remember ANY Tailwind naming, just keep the [Tailwind Cheat Sheet](https://nerdcave.com/tailwind-cheat-sheet) opened.
```

Comment:
There's also a [Raycast addon](https://www.raycast.com/vimtor/tailwindcss), which could come in helpful :)